crates/cosmos/cosmos-chain-components/src/impls/queries/received_packet.rs

```diff
@@ -1,19 +1,21 @@
+use cgp_core::prelude::*;
 use hermes_relayer_components::chain::traits::queries::packet_is_received::ReceivedPacketQuerier;
 use hermes_relayer_components::chain::traits::types::ibc::HasIbcChainTypes;
-use ibc_relayer::chain::handle::ChainHandle;
+use ibc_proto::ibc::core::channel::v1::query_client::QueryClient as ChannelQueryClient;
 use ibc_relayer::chain::requests::QueryUnreceivedPacketsRequest;
 use ibc_relayer_types::core::ics04_channel::packet::Sequence;
 use ibc_relayer_types::core::ics24_host::identifier::{ChannelId, PortId};
 
-use crate::traits::chain_handle::HasBlockingChainHandle;
+use crate::traits::grpc_address::HasGrpcAddress;
 
 pub struct QueryReceivedPacketWithChainHandle;
 
 impl<Chain, Counterparty> ReceivedPacketQuerier<Chain, Counterparty>
     for QueryReceivedPacketWithChainHandle
 where
     Chain: HasIbcChainTypes<Counterparty, ChannelId = ChannelId, PortId = PortId>
-        + HasBlockingChainHandle,
+        + HasGrpcAddress
+        + CanRaiseError<eyre::Report>,
```

Comment:
We can check what error is returned from calls like `ChannelQueryClient::connect`, and add the `CanRaiseError` constraint for that specific error here.

Reply:
it actually return a `tonic::transport::Error`, so I guess should ` CanRaiseError<eyre::Report>`  be enough?

Reply:
In principle, yes. However, `eyre::Report` is too general and obscures details about what kind of error it is. If we add the constraint `CanRaiseError<tonic::transport::Error>`, the chain context can easily choose to handle it in different way, e.g. [marking the error as retryable](https://github.com/informalsystems/hermes-sdk/blob/main/crates/cosmos/cosmos-relayer/src/impls/error.rs#L128-L129).

In addition, the use of CGP and the `CanRaiseError` trait makes it easy to raise any custom error to the chain context. Hence, we prefer to raise errors that are as specific as possible.